openmls/src/schedule/mod.rs

```diff
@@ -745,6 +757,57 @@ impl ExporterSecret {
     }
 }
 
+/// A secret that we can derive secrets from, that are used outside of OpenMLS.
+/// In contrast to `[ExporterSecret]`, the `[ApplicationExportSecret]` is not
+/// persisted. It can be deleted after use to achieve forward secrecy.
+#[cfg(feature = "extensions-draft")]
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[cfg_attr(any(test, feature = "test-utils"), derive(PartialEq))]
+pub struct ApplicationExportSecret {
+    secret: Secret,
+}
+
+#[cfg(feature = "extensions-draft")]
+impl ApplicationExportSecret {
+    /// Derive an `ExporterSecret` from an `EpochSecret`.
+    fn new(
+        crypto: &impl OpenMlsCrypto,
+        ciphersuite: Ciphersuite,
+        epoch_secret: &EpochSecret,
+    ) -> Result<Self, CryptoError> {
+        let secret =
+            epoch_secret
+                .secret
+                .derive_secret(crypto, ciphersuite, "application_export")?;
+        Ok(ApplicationExportSecret { secret })
+    }
+
+    /// Derive a `Secret` from the exporter secret. We return `Vec<u8>` here, so
+    /// it can be used outside of OpenMLS.
+    pub fn derive_exported_secret(
+        &self,
+        ciphersuite: Ciphersuite,
+        crypto: &impl OpenMlsCrypto,
+        label: &str,
```

Comment:
Maybe add something to the doc comment that the `label` should be unique in the application context.